be2-scala/src/main/scala/ch/epfl/pop/model/objects/Channel.scala

```diff
@@ -20,6 +20,20 @@ final case class Channel(channel: String) {
     case _ => None
   }
 
+  /**
+   * Extract the laoId from a channel name (even though it might be in the middle)
+   *
+   * @return an array of bytes corresponding to the decoded laoId or None if an error occurred
+   */
+  def decodeChannelLaoId: Option[Array[Byte]] = channel match {
```

Comment:
I believe that it does not really change much in this case, apart from saving us some .getBytes in other parts of the code since it's directly a Byte array, since the return values are mostly used for database accesses anyway

Reply:
But if you really believe that it would be a relevant change I'll make the modifications in the code

Reply:
I strongly second @BadrTad 's remark.

If I look at the usage of this function, it seems that the use of a `Array[Byte]` has already led to mistakes.
The bytes are being written out directly as a string (which is most likely wrong) , sometimes re-encoded as Base64Data,  etc. 
If you use the `Base64Data` abstraction, you'll probably save yourself from some bugs 😄 

Essentially, it's more or less the difference between a weak-typed and strongly-typed language. You can do the same thing in both, and you can do the same things with `Byte` and `Base64Data`, but the former requires you to be A LOT more careful whereas the latter keeps a convenient abstraction which prevents you from making mistakes.

This is also why we introduced these types, because at the beginning everything was a "string", and you wouldn't notice if swapping a base64 field with, say, a channel. 😉

Reply:
Added this!